// crutch-vue/js/data.ts
import { ref } from 'vue';
import { formatDateMMDDYYYY } from '../utilities/DateTimeFormat';

export const TagData: Record<string, string[]> = {
    Acoustic: ["Any*", "Treatment"],
    Advertising: [
        "Any*",
        "Background",
        "Banner",
        "Catalog",
        "Contest",
        "Free Shipping",
        "Promo",
        "Sweepstakes",
    ],
    Amplifier: [
        "Any*",
        "Built-In",
        "Commercial",
        "Headphone",
        "Integrated",
        "Mono",
        "Multi-room",
        "Power",
    ],
    Antenna: ["Any*", "TV"],
    "Art Director Rating": ["Any*", "Favorite"],
    "ATV/UTV": [
        "Any*",
        "Amplifier",
        "Audio",
        "Head Unit",
        "Installation",
        "Light",
        "Sound Bar",
        "Sound System",
        "Speaker",
        "Stereo",
        "Subwoofer",
    ],
    Audio: [
        "Any*",
        "Commercial",
        "Computer",
        "Control Surface",
        "Interface",
        "Mixer",
        "Multi-room",
        "Portable",
        "Recorder",
        "System",
    ],
    "Bike Bicycle": [
        "Any*",
        "Carrier",
        "Computer",
        "GPS",
        "Light",
        "Mount",
        "Safety",
    ],
    Bluetooth: ["Any*", "Adapter", "Remote"],
    "Boon Rawd Brewery Singha Complex partner story case study corporate Bangkok Thailand Mahajak Develop":
        [
            "Any*",
            "300 amplifiers.JBL LSR305 Mkll studio monitors.Soundcraft Ui Series mixers",
        ],
    "BTS (Behind-the-Scenes)": [
        "Any*",
        "Bow Wow Walk",
        "C.A.R.T. (Charlottesville Area Riding Therapy)",
        "CASPCA",
        "Chalk Wall",
        "Charity",
        "Charity Auction",
        "Chili Cook Off",
        "Christmas Cubicles",
        "Coat Drive",
        "Cookie Decorating",
        "Cookout",
        "Day of Caring",
        "DC Construction",
        "Employee Appreciation Day",
        "Employee of the Month",
        "Halloween",
        "Heart Walk",
        "Howl-o-ween Spooktacular",
        "Independence Day",
        "Pet Food Donation",
        "Picnic",
        "Puppylooza",
        "Retirement Party",
        "SPCA Adoptathon",
        "St. Pawtricks",
        "Tech for Troops",
        "Tom Sox Night",
        "Tomato Tasting",
        "Ugly Sweaters",
    ],
    Buick: ["Any*", "Enclave"],
    Cable: [
        "Any*",
        "A/V",
        "Adapter",
        "Audio",
        "Connection",
        "Ethernet",
        "HDMI",
        "In-wall",
        "Label",
        "Microphone",
        "Power",
        "Power Extension",
        "Speaker",
        "Subwoofer",
        "USB",
    ],
    Camera: [
        "Any*",
        "Action",
        "Backup",
        "Bag",
        "Blind Spot",
        "Dashcam",
        "Digital",
        "Digital SLR",
        "Indoor",
        "Infrared",
        "Instant",
        "Mirrorless",
        "Motion-activated",
        "Mount",
        "Outdoor",
        "Pet",
        "Point-and-Shoot",
        "Professional",
        "Rear View",
        "Security",
        "Surveillance",
        "Tripod",
        "Video",
        "Waterproof",
    ],
    Car: [
        "Any*",
        "Amplifier",
        "Camera",
        "Cargo",
        "Carrier",
        "Console",
        "Controls",
        "Head Unit",
        "Kick Panel Enclosures",
        "Light",
        "Location",
        "Microphone",
        "Monitor",
        "Navigation",
        "Phone Charger",
        "Rack",
        "Radio",
        "Rain Guard",
        "Rear View Mirror",
        "Receiver",
        "Remote Start",
        "Speaker",
        "Stereo",
        "Subwoofer",
        "Tool",
        "Tweeter",
        "Wiring",
    ],
    "Car Location": [
        "Any*",
        "Behind Seat",
        "Between Front Seats",
        "Cargo Area",
        "Center Dash",
        "Center Panel",
        "Console",
        "Corner Dash",
        "Custom Kick Panel",
        "Dash",
        "Exterior",
        "Floor Below Center of Dash",
        "Front Door",
        "Front Pillar Grab Handles",
        "Front Seat",
        "Glovebox",
        "Headlight",
        "Headrest",
        "In Cab Corners",
        "Interior",
        "Kick Panel",
        "Knee Panel",
        "License Plate",
        "Power Port",
        "Rear Deck",
        "Rear Deck Corner",
        "Rear Dooe",
        "Rear Door",
        "Rear Hatch",
        "Review Mirror",
        "Roll Bar",
        "Roof",
        "Seat",
        "Second/Third Row Seats",
        "Side Mirror",
        "Side Panel - Far-rear",
        "Soundbar",
        "Steering Column",
        "Steering Wheel",
        "Tail Door",
        "Tailgate",
        "Trunk",
        "Under",
        "Vent",
        "Wheel",
        "Windshield",
    ],
    "Car Year": [
        "Any*",
        "1967",
        "1988",
        "1994",
        "1998",
        "2000",
        "2002",
        "2005",
        "2006",
        "2007",
        "2008",
        "2009",
        "2010",
        "2011",
        "2012",
        "2013",
        "2015",
        "2016",
        "2017",
        "2018",
        "2019",
    ],
    Catalog: [
        "Any*",
        "April",
        "Aug.-Sept.",
        "Cover",
        "Fall",
        "Holiday",
        "July-Aug.",
        "Late",
        "Mar.-April",
        "May-June",
        "Sept.-Oct.",
        "Spring",
        "Summer",
        "Winter",
    ],
    "Charging Charger": ["Any*", "Dock", "Station"],
    Chevrolet: [
        "Any*",
        "Camaro",
        "Camry",
        "Caprice",
        "Colorado",
        "Equinox",
        "S-10",
        "Silverado",
        "Tahoe",
        "Traverse",
    ],
    "Crutchfield Logo": ["Any*", "Crutchfield.com", "Unleashed"],
    Customer: ["Any*", "Photo", "Testimonial"],
    Damn: ["Any*", "Daniel"],
    Department: [
        "Any*",
        "Accounting",
        "Admin",
        "Advisor",
        "Art Director",
        "AV Design",
        "Blacksburg",
        "Blog",
        "Brand Manager",
        "Business Development",
        "Buyer",
        "Car",
        "Catalog",
        "Commercial Design",
        "Contact Center",
        "Content Manager",
        "Content Officer",
        "Corporate Sales",
        "Creative",
        "Customer Care",
        "Customer Communications",
        "Customer Experience",
        "Customer Support",
        "Data Entry",
        "Designer",
        "Digital Imaging",
        "Director",
        "Distribution",
        "Editor",
        "Email",
        "Employee Development",
        "Employee Relations",
        "Employee Services",
        "Engineering",
        "Executive",
        "Financial",
        "Founder & CEO",
        "Fulfillment",
        "Home",
        "Home Delivery",
        "HQ Charlottesville",
        "Human Relations HR",
        "Information Technology IT",
        "Installer",
        "Intern",
        "Internet",
        "Inventory Handler",
        "Learning Content",
        "Maintenance",
        "Manager",
        "Marine",
        "Marketing",
        "Materials",
        "Merchandising",
        "Mobile",
        "New Media",
        "Operations",
        "Order Processors",
        "Packing",
        "Pet Tech",
        "Photography",
        "Print",
        "Product Research",
        "Programmer",
        "Reas Ford",
        "Receiving",
        "Replenishment",
        "Research",
        "Residential Design",
        "Retail",
        "Returns",
        "RS11 Charlottesville",
        "RS12 Harrisonburg",
        "Sales",
        "Sales Management",
        "Security",
        "Shipping",
        "Social Media",
        "Spanish Team",
        "Special Inventory",
        "Strategic Planning & Management",
        "Strategic Research",
        "Studio Manager",
        "Supervisor",
        "SWVA - Norton",
        "Team Leader",
        "Tech Support",
        "Testing",
        "Trainee",
        "Training",
        "UX Designer",
        "UX Manager",
        "Vehicle Integration",
        "Vice President",
        "Video",
        "Warehouse",
        "Web Order",
        "Website",
        "Workforce Management",
        "Writers",
    ],
    Diagram: ["Any*", "Wiring"],
    "Digital-to-Analog Converter (DAC)": [
        "Any*",
        "Desktop USB",
        "Headphone",
        "Portable USB",
    ],
    Dodge: ["Any*", "Charger", "Dakota", "Durango", "Ram 3500"],
    Drone: ["Any*", "Commercial", "Underwater"],
    Employees: [
        "Any*",
        "Aaron Collier",
        "Aaron Raymond",
        "Adam BeVille",
        "Adam JR Stoffel",
        "Adam Taylor",
        "Adrianna Audri Fields",
        "Adrienne Cupp",
        "Alan Sites",
        "Alan Tiangson",
        "Aleta Childress",
        "Alex Mays",
        "Alexander Alex Belgard",
        "Alexander Alex Byrd",
        "Alexander Alex Taylor",
        "Alexander Lex Hrabe",
        "Alexandra Alex Bugge",
        "Alexis Goodwin",
        "Alfonso Alvarado",
        "Alicia Marshall",
        "Allen Diets",
        "Allison R Brewer",
        "Allison Stough",
        "Amanda Manners",
        "Amanda Sturgill",
        "Amy Lenert",
        "Andrea Athena Hightower",
        "Andrew Andy Washburg",
        "Andrew Drew Wycoff",
        "Andrew Mick Schroeder",
        "Angela Angie Carle",
        "Angela Angie Price",
        "Angie Eubanks",
        "Annissa Carmitchel",
        "Anthony Malave",
        "Anthony Malave Colon",
        "Anthony Moler",
        "Anthony Tony Dino Laquintano",
        "Arielle Ember Smith",
        "Arthur Artie Arico",
        "Ashley Moore",
        "Augustus Gus Anderson",
        "Autumn Brooke Ring",
        "Barbara Bobbi Herron",
        "Barbara Payne",
        "Barry Wilson",
        "Benjamin Ben Sams",
        "Benjamin May",
        "Benn Benjamin Grant",
        "Beverly Lynn Herring",
        "Bjorn Sorensen",
        "Blake Livesay",
        "Bo Adams",
        "Brad Collier",
        "Bradley Dexter Winebergar",
        "Bradley Stapleton",
        "Brandi Blair Kilgore",
        "Brandon Bevins",
        "Brandon Early",
        "Brandon Johnson",
        "Brandon Perry",
        "Bree Baker",
        "Brett Fitzgerald",
        "Brett Hamm",
        "Brett McDaniel",
        "Brian Grant",
        "Brian Madison",
        "Brian Ryan Shield",
        "Britney Brit Huff",
        "Brittany Brit Keller",
        "Brooks Wellman",
        "Bruce Southall",
        "Bruce Zac Bannan",
        "Bryan Langley",
        "C. Mark Garwood",
        "Caressie Case Reasor",
        "Carl Matthews",
        "Carlos Acosta Espada",
        "Carmen Schoelkopf",
        "Casey Cosmo Cramer",
        "Cecil Boggs",
        "Charles Chad Jones",
        "Charles Charlie Jones",
        "Charles Charlie Pastorfield",
        "Charles Chuck Bowen",
        "Charles Fontaine",
        "Charles Luke Anderson",
        "Charlie Jones",
        "Chris Colt Henofer",
        "Chris Groseclose",
        "Chris Hall",
        "Chris Martin",
        "Chris Mullins",
        "Chris Powell",
        "Chris Tyler Kilgore",
        "Christian Chris Lilley",
        "Christian Jane Fraley",
        "Christina Dorsey",
        "Christina L Lucy Robinette",
        "Christina Leiby",
        "Christopher Chris Baker",
        "Christopher Chris Barefoot",
        "Christopher Chris Caracofe",
        "Christopher Chris Colt Henofer",
        "Christopher Chris Groseclose",
        "Christopher Chris Henofer",
        "Christopher Chris Meadows",
        "Christopher Chris Robert Greear",
        "Christopher Chris Wade Heffner",
        "Christopher Chris Welte",
        "Christopher Dante Keenan",
        "Christopher Gage Passantino",
        "Christopher Gus Cramer",
        "Christopher Hall",
        "Christopher Mays",
        "Christopher Wade Heffner",
        "Christy Tromey",
        "Chuck Bowen",
        "Ciara Fowler Heffner",
        "Ciara Heffines",
        "Ciara Heffner",
        "CJ McCoy",
        "Clint Stilwell",
        "Cody Childress",
        "Cole Fuller",
        "Colin Cal Mullins",
        "Connie Ferency-Viars",
        "Connor Dwight Lilley",
        "Conrad Drennon",
        "Corey Haga",
        "Corey Ray Sanchez",
        "Corky Schoonover",
        "Corwin Corey Haga",
        "Cory Colby Freeland",
        "Cory Teitelbaum",
        "Courtney Scarlett Young",
        "Cris Baker",
        "Crystal Parker",
        "Crystal Salas",
        "Curtis CJ McCoy",
        "Curtis Prince",
        "Dakota Vandergriff",
        "Dale Sturgill",
        "Dan Hodgson",
        "Daniel Abrams",
        "Daniel Dan Hodgson",
        "Daniel Dan Peterson",
        "Daniel Deal",
        "Daniel Miller",
        "Daniel Smitty Minahan",
        "David Dave Bar",
        "David Dave Brown",
        "David Dave Campbell",
        "David Dave Delamere",
        "David Dave Nameth",
        "David Dave Weisman",
        "David Larry Scott",
        "David Lee",
        "David Sturgill",
        "Dawn Dickenson",
        "Deanna Gould",
        "Deborah Debbie McPhillips",
        "Debra Deb Gilbert",
        "Debra Deb Molly Gilbert",
        "Deianeria Deia Eva Zukowski",
        "Deianeria Deia Zukowski",
        "Demetrius Dee Turner",
        "Devin Adkins",
        "Devin McDaniel",
        "Devin Morris",
        "Diane Cornett",
        "Dominic Dom Devito",
        "Don Minson",
        "Douglas Dallas Simon",
        "Douglas Doug Drummer Liggins",
        "Douglas Doug Schmelter",
        "Dulcie Jax Fields",
        "Duncan Ferency-Viars",
        "Dustin Hall",
        "Edward Ed Stockwell",
        "Edward Ned Oldham",
        "Elijah Romaine",
        "Ella Reed",
        "Elysia Ella Reed",
        "Emily Montgomery",
        "Emily Sohr",
        "Emma Jackson",
        "Emmanuel Kartey",
        "Eric Angevine",
        "Eric Battle",
        "Eric Huffman",
        "Eric Huffy Huffman",
        "Eric Jackman",
        "Eric Smith",
        "Erica Hayes",
        "Erik Otis Swartzwelder",
        "Erin Batten",
        "Erin Dolly Batten",
        "Erin Dolly Graves",
        "Ethan Meyer",
        "Eugene Gene Chan",
        "Evan Swecker",
        "Faith Hubbard",
        "Ford Stallard",
        "Francis McCauley",
        "Gabe Gus Strauss",
        "Garrett Bowman",
        "Gary Gibbs",
        "Gary Ziggy McLaren",
        "George Wade",
        "Geran Steljes",
        "Glenn Norm Hawes",
        "Goebel Sean Levi Wright",
        "Gregory Eddie Telep",
        "Gregory Greg Morris",
        "Group",
        "Hailey Sheehan",
        "Hank Cook",
        "Hank Hubbard",
        "Hannah England",
        "Hannah Gilliam",
        "Hannah Pham",
        "Harriett Schultz",
        "Heather Merchant",
        "Heidi Hutson",
        "Helen Earley",
        "Helen Myers",
        "Holly HB Roberts",
        "Hope Newberry",
        "Howard Riddle",
        "Jacob Alston",
        "Jacob Alstow",
        "Jacob Malamphy",
        "James Barber",
        "James Darren Stanley",
        "James Jalen Parrish",
        "James Jim Ralston",
        "James Jim Richardson",
        "James Jim Sonny Ashley",
        "James Jimmy Baranik",
        "James Jimmy Wynne",
        "Jane Fraley",
        "Janet Janney Stout",
        "Jarathan Neal Cook",
        "Jarrod Williams",
        "Jason Fletcher",
        "Jason Flowers",
        "Jason Hockman",
        "Jason Jay Hale",
        "Jason Jay McCartney",
        "Jason Jayce Sanchez",
        "Jason Lee",
        "Jason Manny Hall",
        "Jason Rex Heuer",
        "Jason Sanchez",
        "Jason Showalter",
        "Jason Stanford",
        "Jaye Light",
        "Jeanne Jill Smith",
        "Jeanne Smith",
        "Jeff Miller",
        "Jeff Rusty Powell",
        "Jeffrey Hancock",
        "Jenna Cooke",
        "Jennifer Jennie Vines",
        "Jennifer Jenny Phan",
        "Jensen Malec",
        "Jesse Boyd",
        "Jessica Jess Greer",
        "Jessica River Hill",
        "Jessica Stull",
        "Jim Richardson",
        "JimBob",
        "Jimmy Baranik",
        "JoAnna Penny Kennedy-Booher",
        "Joel Briggs",
        "Joella Leia Sloand",
        "John Bob Hoy",
        "John Cooley",
        "John Curtis Prince",
        "John Doucette",
        "John J. H. Stoll",
        "John Loehr",
        "John Matthias",
        "Jolene Ashley",
        "Jon Paulette",
        "Jon Sykes",
        "Jonathan Gordo Gordon",
        "Jonathan Jon Paulette",
        "Jonathan Warner",
        "Jordan Coley",
        "Jordan Greenstone",
        "Joseph Jac Collier",
        "Joseph Joe Joey Morton",
        "Joseph Joey Morton",
        "Joseph Lex Trust",
        "Joseph Salyers",
        "Josh Dierauer",
        "Josh Morgan Dierauer",
        "Joshua JC Carroll",
        "Joshua Josh Khran",
        "Joshua Josh Krahn",
        "Joshua Josh Kramer Crane",
        "Joshua Josh Rollins",
        "Juan Ascoli",
        "Jude DeFrank",
        "Julia Phillips",
        "Julie Crum",
        "Julie Govan",
        "Julie Wright",
        "Justin Gibbons",
        "Justin Remy Skidmore",
        "Juwhan Lee",
        "Kaitlyn Aria Rose",
        "Kate Olsen",
        "Kate Pugh",
        "Katherine Kate Kay Robertson",
        "Katherine Kate Robertson",
        "Kathryn DeFrank",
        "Kathy Covington",
        "Kayla Ashley",
        "Kayla Phillips",
        "Kayla Piper Ashley",
        "Kelli Cooper",
        "Kelly Greenstone",
        "Kenas Scot Adams",
        "Kendra Shelton Morris",
        "Kenneth Gibson",
        "Kenneth Ken Huck",
        "Kenneth Ken Spike Nail",
        "Keri Crum",
        "Kinsey Page",
        "Korey Archer Adkins",
        "Kristian Kris Konkle",
        "Kristina Johnson",
        "Kristina Krissy Pitts",
        "Kristina Veney",
        "Kristine GÇÿKatGÇÖ Olaes",
        "Kristine Kat Olaes",
        "Kristopher Kris Kirtley",
        "Kurt Brad Arnold",
        "Kurt Goodwin",
        "Kyle Ezer",
        "Kyle Hamilton",
        "Landon Rody",
        "Lane Daniel",
        "Larry Allen Deets",
        "Larry Allen Diets",
        "Larry McGough",
        "Laura Collumus Davis",
        "Laura Lola Bush",
        "Lauren Evans",
        "Leah Francis",
        "Leslie Davidson",
        "Lindsey Houchens",
        "Lindsey Luna Houchens",
        "Lisa Pushard",
        "Logan O'Neill",
        "Logan Osbourne",
        "Loren Barstow",
        "Lori Campbell",
        "LT Gillmer",
        "Lyndsey Quinn Harlow",
        "Lynn Herring",
        "Mackenzie Carter Carter",
        "Maggie Flinn",
        "MaKenna Engle",
        "Malkam Cam Martin",
        "Malkam Martin",
        "Mandy Blanton",
        "Maria Mantilla",
        "Marian Ivy Howard",
        "Marie Barfield",
        "Marigold Haske",
        "Marisa Page",
        "Mark Casper",
        "Mark Craddock",
        "Mark Critchfield",
        "Mark Sparky Conner",
        "Mark Tony Bolling",
        "Marlon Cerritos",
        "Marshall Chase",
        "Martin Brady Norman",
        "Martyn Kyle",
        "Marvin Lynne Stallard",
        "Mathes Glymph",
        "Mathilda McCommon",
        "Matthew Forrest Grubb",
        "Matthew Grubb",
        "Matthew Matt Baskin",
        "Matthew Matt Freeman",
        "Matthias Theo John",
        "Megan Duffy",
        "Megan Tatum Redling",
        "Melissa McMullan",
        "Melissa Missy McMullan",
        "Michael Crenshaw",
        "Michael Hoffman",
        "Michael Jason Stanford",
        "Michael Lee Morris",
        "Michael Lloyd Herron",
        "Michael Mike Bost",
        "Michael Mike Brown",
        "Michael Mike Colley",
        "Michael Mike Sokolowiski",
        "Michael Stanford",
        "Michelle Hurd",
        "Mike Cain",
        "Mike Morris",
        "Morgan Harley Conner",
        "Morgan Mac Clark",
        "Morgan Mack Clark",
        "Nancy Janelle Wood",
        "Nathan Rocky Carpenter",
        "Nathan Sturgill",
        "Nathaniel D Nate Smith",
        "Nathaniel Smith",
        "Nick DeFrank",
        "Nicole Baisden",
        "Nicole Glymph",
        "Nicole O'Hearn",
        "Ninai Koplan",
        "Noah Childress",
        "Norman Norm Hawes",
        "Oakley Salyers",
        "Oday Yonus",
        "Oddie Lee Bernie Carneal",
        "Oddie Lee Carneal",
        "Olga Beltre",
        "Oliver Minahan",
        "Oliver Rutter",
        "Olivia McDaniel",
        "Ovido Alfonzo Alvarado-Zamora",
        "Pat McCrystal",
        "Patricia Trish England",
        "Patrick Allen",
        "Patrick Pat Dale McPhillips",
        "Patrick Pat Howard",
        "Patrick Pat McPhillips",
        "Patrick Spraker",
        "Paul Jeremy Sechler",
        "Paul Kavanaugh",
        "Paul Miller III",
        "Paula Parker Taylor",
        "Peggy Lawrence",
        "Penny Booher",
        "Peter Bramley",
        "Peter Logan",
        "Peter Pete Clarence Mullen",
        "Peter Pete Mullen",
        "Peter Reid Barcia",
        "Phillip Abbott",
        "Phillip Phil Jones",
        "Portrait",
        "Rachel Blakistone",
        "Rachel Showalter",
        "Raina Greer",
        "Ralph Graves",
        "Randy Conley",
        "Randy Vasquez",
        "Rashad Mosley",
        "Ray Salinas",
        "Raymond Pete Ducharme",
        "Rayna Pierce",
        "Ricardo Nunez",
        "Rich Stavitski",
        "Richard Rick Wright",
        "Rob Parker",
        "Rob Richmond",
        "Robbie Shifflett",
        "Robert Bob Bihlmayer",
        "Robert Buck Pomerantz",
        "Robert Chris Greear",
        "Robert Ferency-Viars",
        "Robert Opie Jr.",
        "Robert Rob Bob Richmond",
        "Robert Rob Parker",
        "Robert Rob Tran",
        "Robert Rusty Powell",
        "Robin Banaciski",
        "Rocco Bernardi",
        "Ron Histand",
        "Roy RJ Robbins",
        "Russell Russ Brill",
        "Ryan Oliver Rutter",
        "S. Skip Payne",
        "Sabra Bolling",
        "Said Benabed",
        "Samuel Sam Denham",
        "Samuel Sam Gilliam",
        "Samuel Stan Adams",
        "Sandra Sandy Bahorinsky",
        "Sarah Benning",
        "Savannah Vanna Smith",
        "Scheline Crutchfield",
        "Scott Anderson",
        "Scott Ferguson",
        "Scott Thompson",
        "Sean Murphy",
        "Sebastian Seb Stephenson",
        "Shaji Raghavan",
        "Shane Allen",
        "Shaun Matt Bevins",
        "Shawn Arnold",
        "Shawn Dunaway",
        "Shawni Kilgore",
        "Shena Culbertson",
        "Sherry Wright",
        "Shu-Jen SJ Chen",
        "Sierra Deane",
        "Sierra Gardner",
        "Simon Kendrick",
        "Siobhan Kent",
        "Sophie Bromberger",
        "Stacey Batten",
        "Stafford Tyler Woody Woodall",
        "Stephanie Bell",
        "Stephanie Giba",
        "Stephanie Kester",
        "Stephanie Nicole Wingler",
        "Stephen Davis",
        "Stephen Hunter Hamilton",
        "Stephen Steve Kindig",
        "Stephen Steve Pippin",
        "Stephen Stone Wilson",
        "Steve Silberman",
        "Steven Daniel",
        "Suhrab Nizaam",
        "Susan Kerry",
        "Svetlana Alana Rivas Lopez",
        "Tamara Doucette",
        "Tamara Tami Walker",
        "Tammi Walker",
        "Tammie Phillips",
        "TaQuan Pace",
        "Tara Wisnewski",
        "Tara Yoder",
        "Taylor Kara Price-O'brien",
        "Taylor Stanley",
        "Teddy Campbell",
        "Terry Breedon",
        "Theresa Weidman",
        "Thomas Calvin Lowe",
        "Thomas Hogan Czysz",
        "Thomas Lowe",
        "Thomas Tom Greer",
        "Thomas Tommy Kinstle",
        "Tiffany Crabtree",
        "Tim Freeman",
        "Timothy Root",
        "Timothy Tim Hensen",
        "Timothy Tim Taylor",
        "Travis Belcher",
        "Travis Bingler",
        "Travis Giles",
        "Travis Paradis",
        "Treva Allen",
        "Trevon Johnson",
        "Trevor Allen",
        "Tylaher Dailey",
        "Tyler Bare",
        "Valentin Zaldivar",
        "Vernard Scruggs",
        "Veronica Atakpa",
        "Verrard Scrubbs",
        "W Mackie Wallenborn",
        "Wallace Butch Robertson",
        "Waye Todd Peery",
        "Wayne Wilberger",
        "Wendetta Wendy P Dowdell",
        "Wendy Carlson",
        "Wesley Taylor Polly II",
        "Wesley Wes Strouth",
        "William Bill Crutchfield",
        "William Buzz White",
        "William Huff",
        "William King",
        "William Rob Bryce Pitts",
        "William Rob Pitts",
        "William Will Crutchfield Jr.",
        "Winston Hughes",
        "Zach Zimet",
        "Zachary Nemo Hill",
        "Zachary Phillips",
        "Zachary Zach Barto",
        "Zachary Zach Bear Kinn",
        "Zachary Zach Kinn",
        "Zachary Zak Billmeir",
        "Zebulun Cody Estep",
    ],
    Factory: [
        "Any*",
        "Amplifier",
        "Backup Camera",
        "Head Unit",
        "Speaker",
        "Subwoofer",
        "Tweeter",
    ],
    Fiat: ["Any*", "500"],
    Ford: [
        "Any*",
        "Escape",
        "Expedition",
        "Explorer",
        "F-150",
        "Focus",
        "Mustang",
        "Ranger",
        "Raptor",
        "Sport Trac",
        "Super Duty",
        "SuperCab",
    ],
    Furniture: [
        "Any*",
        "Audio Cabinet/Rack",
        "Entertainment",
        "Office",
        "Shelving",
        "Wall-mountable",
    ],
    Gender: ["Any*", "Female", "Male"],
    GMC: ["Any*", "Acadia", "Canyon", "Sierra", "Terrain"],
    Golf: [
        "Any*",
        "Audio",
        "Cart",
        "Dash",
        "GPS",
        "Launch Monitor",
        "Rangefinder",
        "Simulator",
        "Stereo",
    ],
    GPS: ["Any*", "Car", "In-dash", "Off-road", "Portable"],
    Headphones: [
        "Any*",
        "Acoustic",
        "Earbud",
        "In-ear",
        "Neckband",
        "Noise-cancelling",
        "On-ear",
        "Over-the-ear",
        "Sports",
        "Stereo",
        "TV",
    ],
    Home: [
        "Any*",
        "Audio",
        "Automation",
        "Computer Speaker",
        "Doorbell",
        "Lock",
        "Monitoring",
        "Powered System",
        "Receiver",
        "Remote Control",
        "Seating",
        "Security",
        "Smart",
        "Speaker",
        "Stereo",
        "Theater",
        "Thermostats",
        "Video Doorbell",
    ],
    Honda: [
        "Any*",
        "Accord",
        "Civic",
        "Civic LX",
        "Del Sol",
        "Fit",
        "Hybrid",
        "VTX 1800C",
    ],
    Install: ["Any*", "Car", "Commercial", "Home", "Marine"],
    Integration: ["Any*", "Android"],
    Jeep: [
        "Any*",
        "Cherokee",
        "Compass",
        "Gladiator",
        "Grand",
        "Rubicon",
        "Sport",
        "Unlimited",
        "Wrangler",
    ],
    Kayak: ["Any*", "Light", "Power"],
    Kia: ["Any*", "Forte", "Sportage"],
    Lens: ["Any*", "DSLR", "Mirrorless", "Telephoto", "Zoom"],
    Location: [
        "Any*",
        "At Work",
        "Boat",
        "Call Center",
        "CASPCA",
        "Christiansburg Testing Facility",
        "Distribution Center",
        "Employee's Home",
        "Home Labs",
        "HQ",
        "In Car",
        "Indoors",
        "Install Lab",
        "Lab",
        "New York",
        "On a Boat",
        "Outdoors",
        "Reas Ford",
        "RS11 Charlottesville",
        "RS12 Harrisonburg",
        "Sales Floor",
        "Studio",
        "SWVA",
        "The House",
        "Vendor Training Room",
        "Video Studio",
        "Woofer Park",
    ],
    Logo: [
        "Any*",
        "2nd Life",
        "Actable App",
        "Android",
        "aptX",
        "AT&T",
        "Audyssey",
        "AUPEO",
        "Authorized Retailer",
        "BetterBusinessBureau",
        "Bill Me Later",
        "BizRate",
        "Blockbuster",
        "Blu-ray Disc",
        "Buy 1 Get 1 BOGO",
        "CASPCA",
        "CES",
        "CExchange",
        "Crutchfield",
        "Daily Progress Readers Choice",
        "Digital Cable Ready",
        "DIRECTV",
        "DMA",
        "Dolby",
        "Dreamworks",
        "DSS",
        "DTS",
        "DVD",
        "Energy Guide",
        "ESPN",
        "Extreme Home Makeover",
        "Facebook",
        "Fandango Now",
        "Forbes",
        "Grooveshark",
        "HD DVD",
        "HD Radio",
        "HDMI",
        "HDtracks",
        "Heart Gala",
        "Hulu",
        "IDme",
        "Inrix",
        "InstallItYourselfandSave",
        "Internet Retailer",
        "iTunes",
        "JD Power",
        "JMU",
        "Kiplinger",
        "Kleer",
        "Last.fm",
        "MECP",
        "MediaFly",
        "Metra",
        "Meze Audio",
        "MHL",
        "Midland",
        "MiniDisc",
        "Mirage",
        "MITSUBISHI",
        "Mobile Toys",
        "Mofi",
        "MOG",
        "Mohu",
        "Monitor Audio",
        "Monster Cable",
        "Morel",
        "MQA",
        "Mr.Speakers",
        "MTX",
        "Multi Wedge",
        "Musical Fidelity",
        "MXL",
        "NAD Electronics",
        "NavAtlas",
        "Navigon",
        "NBA",
        "Netflix",
        "Netgear",
        "NHL",
        "Pandora",
        "Payment Option",
        "Pinterest",
        "Play For Sure",
        "Play-Fi",
        "Qello",
        "RCA",
        "Rdio",
        "Recycle Please",
        "samsung",
        "Shure",
        "Slacker",
        "Snapfish",
        "Social Media",
        "SomaFM",
        "Spotify",
        "Super Audio CD",
        "Time Magazine",
        "ting",
        "Tivo",
        "TuneIn",
        "TV Bugs",
        "TV Guide",
        "Twitter",
        "University of Virginia",
        "UPS",
        "USB",
        "V.me",
        "varad",
        "VIA Virginia Institute of Autism",
        "Virginia Business",
        "Vturner",
        "VUDU",
        "WesternUnion",
        "Wifi",
        "YouTube",
    ],
    Map: [
        "Any*",
        "Shipping",
        "To Charlottesville Store RS11",
        "To Harrisonburg Store RS12",
        "To Norton SWVA",
    ],
    Marine: [
        "Any*",
        "Amplifier",
        "Boat Accessories",
        "Chartplotter",
        "Fish Finder",
        "GPS",
        "Lighting",
        "Media Receiver",
        "Radio",
        "Remote",
        "Speaker",
        "Subwoofer",
    ],
    Mazda: ["Any*", "5", "CX", "Navajo"],
    Media: ["Any*", "E-mail", "Social Media"],
    Mercury: ["Any*", "Mariner", "Mountaineer"],
    Mircophone: ["Any*", "Calibration", "Stand", "Wireless"],
    Mixer: ["Any*", "Analog", "Audio", "Commerical", "Digital"],
    Motorcycle: ["Any*", "Audio", "Navigation", "Receiver"],
    Music: ["Any*", "Player", "Streamer", "System"],
    NewBatchTagName: ["Any*", "NewBatchTagValue"],
    NewTagName: ["Any*", "NewTagValue"],
    Nissan: ["Any*", "Juke"],
    "Other Keywords": [
        "Any*",
        "105KWV840B",
        "206CXA1200",
        "714R10B",
        "adventure",
        "Best of CTIA",
        "JBL BassPro Go SUBBPGO SUBBPGOAM BassPro Go",
        "Mark Trista",
    ],
    "Pet Technology": [
        "Any*",
        "Activity Monitor",
        "Anti-Bacterial Water Bowl",
        "Anxiety Relief",
        "Automatic Feeder",
        "Automatic Water Fountain",
        "Barrier",
        "Dog Training",
        "Fountain",
        "Fur Control",
        "GPS",
        "Interactive Toy",
        "Smart",
        "Technology for Cats",
        "Wireless Fence",
    ],
    Pets: [
        "Any*",
        "Abby Sohr",
        "Annie Lenert",
        "Appa Zukowski",
        "Bailey Tromey",
        "Bam Bam",
        "Bert Angevine",
        "Blue Souder",
        "Bo Miller",
        "Buster Bernardi",
        "Cabell McPhillips",
        "Cat",
        "Charlotte Brown",
        "Colbie Huff",
        "Conrad Sohr",
        "Cooper Crutchfield",
        "Cosmo Hamilton",
        "Daisy Boggs",
        "Dog",
        "Felix Zukowski",
        "Fergie Wisnewski",
        "Fern Zukowski",
        "Finn Kindig",
        "Grizzly Norman",
        "Group",
        "Harley Conner",
        "Heiser Pippin",
        "Herbie Miller",
        "Iko Smith",
        "Iris Lowe",
        "Jackson Huff",
        "Janelle Wood",
        "Jason Butler",
        "Jax Huff",
        "Joey Glymph",
        "Juliet Pierce",
        "Junior Barank",
        "Junior LN",
        "Kaiser Arnold",
        "Katey Zukowski",
        "Kaylee Olaes",
        "Leland Zukowski",
        "Lil Bit Pugh",
        "Lincoln Crutchfield",
        "Lucy Sohr",
        "Lulu Govan Hensen",
        "Luna Richmond",
        "Luna Sturgill",
        "Mahogany Koplan",
        "Maisie Showalter",
        "Mara Rutter",
        "Meeka Zukowski",
        "Merf Debit",
        "Milo Showalter",
        "Miyu Harlow",
        "Moose Wisnewski",
        "Mr. President",
        "Ms Ivy Young",
        "Murray Francis-Dimock",
        "Neal Hrabe",
        "Nina Stoll",
        "Nixon Acosta Espada",
        "Ollie Crum",
        "Other Animal",
        "Pattycakes Showalter",
        "Penny Freeman",
        "Porter Freeman",
        "Puppy",
        "Quincy Wood",
        "Radar Sohr",
        "Razzie England",
        "Reese Huff",
        "Ricky Veney",
        "Roxy Baisden",
        "Roxy O'Hearn",
        "Ruger McMullan",
        "Sadie McLeskey",
        "Scherzer Stoffel",
        "Shelby Ascoli",
        "Skellig Tromey",
        "Smokey",
        "Sophia Crum",
        "Sophie Deane",
        "Sully Ramey",
        "Thor Hill",
        "Veronica Sohr",
        "Wendetta Dowdell",
        "Zellie Zimet",
        "Zeppelin McPhillips",
        "Zeus Fraley",
        "Zeus Phan",
        "Zoey Kilgore",
    ],
    "Photo Type": [
        "Any*",
        "Lifestyle",
        "Processed",
        "Silo Silhouette Cut Out",
        "Stock",
        "Test",
        "Vendor",
    ],
    Photography: ["Any*", "Selects"],
    Player: ["Any*", "Blu-Ray", "Blue-Ray", "CD", "DVD"],
    Polaris: ["Any*", "RZR XP 1000"],
    Pontiac: ["Any*", "Firebird"],
    Power: ["Any*", "Pack", "Protection", "Supply"],
    Product: ["Any*", "Group"],
    Projector: ["Any*", "Mount", "Screen"],
    "Public Address (PA)": ["Any*", "Portable System", "Speaker"],
    Radar: ["Any*", "Detectors"],
    Ram: ["Any*", "1500"],
    Receiver: ["Any*", "Multi-room"],
    Remote: ["Any*", "Control", "Start", "Volume Knob"],
    RV: ["Any*", "Audio", "Video"],
    "Satellite Radio": ["Any*", "Dock-and-Play"],
    "Signal Processor": ["Any*", "Equalizer"],
    Smartphone: ["Any*", "Controller", "Storage"],
    "Social Media": [
        "Any*",
        "Facebook",
        "Facebook Live",
        "Instagram",
        "Snapchat",
        "Sweepstakes",
        "Twitter",
    ],
    Speaker: [
        "Any*",
        "2-Way",
        "3-Way",
        "4-Way",
        "Amplified",
        "Bookshelf",
        "Center Channel",
        "Coaxial",
        "Compact",
        "Component",
        "Computer",
        "Desktop",
        "Floor-Standing",
        "In-ceiling",
        "In-wall",
        "Mount",
        "Neck",
        "On-wall",
        "Outdoor",
        "Pendant",
        "Portable",
        "Powered",
        "Satellite",
        "Smart",
        "Stand",
        "Stereo",
        "Surround Sound",
        "System",
        "Tower",
        "Wall Mount",
        "Wire",
    ],
    Stereo: ["Any*", "Home", "Portable", "Powered Speaker", "Preamp", "Receiver"],
    Stock: ["Any*", "Image", "Model"],
    "Stock Image": [
        "Any*",
        "Animal",
        "Background",
        "Fall Autumn",
        "Group",
        "Landscape",
        "Model",
        "Spring",
        "Summer",
        "Winter",
    ],
    Subaru: ["Any*", "BRZ", "Forester", "Outback"],
    Subwoofer: [
        "Any*",
        "Box",
        "Built-In",
        "Compact",
        "Component",
        "Enclosure",
        "Mono",
        "Pendant",
        "Powered",
        "Ulta-Compact",
    ],
    "Surround Sound": ["Any*", "System"],
    Thermostat: ["Any*", "Smart"],
    Toyota: [
        "Any*",
        "4Runner",
        "C-HR",
        "Camry",
        "Highlander",
        "Pickup Truck",
        "Tacoma",
        "Yaris",
    ],
    Turntable: ["Any*", "Cartridges"],
    TV: [
        "Any*",
        "3D",
        "4K",
        "8K",
        "Curved",
        "HD",
        "HDR",
        "LED",
        "OLED",
        "Outdoor",
        "QLED",
        "Smart",
        "Stand",
        "Ultra HD",
        "Wall Mount",
    ],
    "Vehicle Type": [
        "Any*",
        "Classic Car",
        "Compact",
        "Coup",
        "Coupe",
        "Crossover",
        "Hatchback",
        "Hybrid",
        "Off-road",
        "Pickup",
        "Recreational RV",
        "Sedan",
        "Sports",
        "Subcompact",
        "SUV",
        "Trailer",
        "Truck",
        "Utility",
        "Van",
    ],
    Vendor: [
        "Any*",
        "Accele",
        "Acoustic Solutions",
        "Adero",
        "Aiwa",
        "AKG",
        "Alcidae",
        "ALLEN & HEATH",
        "ALPHASONIK",
        "Alpine",
        "Amazon",
        "American International",
        "Anthem",
        "APC",
        "Apogee",
        "Apple",
        "Apple Talk",
        "Aquatic AV",
        "ARCAM",
        "Arista",
        "Artison",
        "Astell & Kern",
        "Astro",
        "Atlas Sound",
        "Audeze",
        "Audio Frog",
        "Audio Solutions",
        "Audio-Technica",
        "Audiobahn",
        "AudioControl",
        "AudioEngine",
        "AudioQuest",
        "AUDIOVOX",
        "Audyssey",
        "August",
        "Auralex",
        "Austere",
        "Autel Robotics",
        "Automatic",
        "Autonet",
        "AVF",
        "Avital",
        "Axxera",
        "B-Tech",
        "Bang & Olufsen",
        "Bass Slammer",
        "BAZOOKA",
        "BDI",
        "BEATS",
        "Behringer",
        "Belkin",
        "BellO",
        "Beltronics",
        "BLAUPUNKT",
        "BLI",
        "Blue Line",
        "Bluesound",
        "Bluetrek",
        "Bojo",
        "Boom Mat",
        "Bora Gear",
        "Borla",
        "BOSE",
        "Bose Pro",
        "BOSS",
        "Boston Acoustics",
        "Bowers & Wilkins",
        "Boyo",
        "Bracketron",
        "Brandmotion",
        "BRAX",
        "Brilliant",
        "Cable Jive",
        "Cambridge Auto",
        "Canon",
        "CarriageWorks",
        "CEA",
        "Center Stage",
        "Cerwin Vega",
        "Channel Master",
        "CinemaNow",
        "Cisco",
        "Clarion",
        "ClearView",
        "Cleer",
        "cobra",
        "Code Alarm",
        "Contour",
        "CRIME GUARD",
        "Crimestopper",
        "Crown",
        "Crux",
        "Ctek",
        "DBX",
        "Definitive Technology",
        "DeLorme",
        "Delphi",
        "Denon",
        "Denon Pro",
        "Devialet",
        "Diamond Auto",
        "Dice Electronics",
        "DJI",
        "DOD Tech",
        "DROPCAM",
        "Dual",
        "Dynamat",
        "Dynaudio",
        "Dyson",
        "Echobox",
        "Eclipse",
        "Eero",
        "EFX",
        "ELAC",
        "Electro Voice",
        "Elite Screens",
        "Energizer",
        "ENERGY",
        "Epson",
        "Escient",
        "Escort",
        "ESS",
        "Eton",
        "Etymotic Research",
        "Farenheit",
        "Fender",
        "FenSens",
        "Fiil",
        "Find My Pet",
        "Fitbark",
        "Flexson",
        "Flo by Moen",
        "FLO TV",
        "Focal",
        "Focusrite",
        "Fortin",
        "Fujifilm",
        "Furman",
        "Furrion",
        "Fusion",
        "Garmin",
        "Gator Cases",
        "Geneva",
        "Gentax",
        "Glasscapes",
        "Google",
        "Google Home",
        "GoPro",
        "GPC",
        "Grace Audio",
        "Grado",
        "HARMON-KARDON",
        "HD Radio",
        "Helix",
        "hella",
        "HEOS",
        "Hertz",
        "HiFiMAN",
        "HIFONICS",
        "Hushmat",
        "Hypertech",
        "I-Way",
        "Ichiban",
        "iDapt",
        "iDatalink",
        "iFetch",
        "ifi",
        "iHome",
        "iLive",
        "Illusion Audio",
        "iLuv",
        "INFINITY",
        "InFocus",
        "InGrid",
        "InstallBay",
        "InstallerNet",
        "Insteon",
        "Integra",
        "iPhone",
        "iPod",
        "Jamo",
        "Jaybird",
        "JBL",
        "JENSEN",
        "Jet Performance",
        "Jiawei",
        "JL Audio",
        "JVC",
        "K&M",
        "K&N",
        "K40",
        "Kanto",
        "KEF",
        "KENWOOD",
        "Key Digital",
        "Kicker",
        "King",
        "Klipsch",
        "Kodak",
        "KRK",
        "LaserShield",
        "Lenmar",
        "LENSBABY",
        "Lexar",
        "Lexicon",
        "LG",
        "Liberty AV",
        "LifeShield",
        "Lightning Audio",
        "Link AKC",
        "Linksys",
        "Littelfuse",
        "Livio",
        "Llumar",
        "Logitech",
        "Lowepro",
        "Lowrence",
        "Lutron",
        "MA Audio",
        "Mackie",
        "Manfrotto",
        "MantelMount",
        "Marantz",
        "Marshall",
        "Martin Logan",
        "Master & Dynamic",
        "Maxell",
        "Maxsa",
        "MB Quart",
        "McIntosh",
        "MECP",
        "Megellan",
        "Memphis Audio",
        "Meridian",
        "Metra",
        "Meze Audio",
        "Midland",
        "Mirage",
        "MITSUBISHI",
        "Mofi",
        "Monitor Audio",
        "Monster",
        "Monster Cable",
        "Morel",
        "Mosconi",
        "Motionwise",
        "Mr.Speakers",
        "MTX",
        "Muse Auto",
        "Music Hall",
        "NAD Electronics",
        "NANUK",
        "NavAtlas",
        "NEST",
        "NHT",
        "Nikon",
        "NILES",
        "Njoie",
        "NOCO",
        "Novation",
        "NuForce",
        "Nuheara",
        "Okki Nokki",
        "Olloclip",
        "Olympus",
        "On-Q",
        "Onkyo",
        "Onkyo Graphics",
        "OnVocal",
        "OPPO",
        "Origin Acoustics",
        "Orion",
        "OSD Audio",
        "Owlcam",
        "PAC",
        "Panamax",
        "PANASONIC",
        "Panavise",
        "Papago",
        "Paradigm Loudspeakers",
        "Parrot",
        "PC Treasures",
        "Peachtree Audio",
        "PearlAuto",
        "Peavey",
        "PEQ",
        "Perf GoGreen",
        "Performance Furniture",
        "Periodic Audio",
        "Pet Acoustics",
        "Petkit",
        "PetSafe",
        "Petzila",
        "Petzila Petzi",
        "PHILIPS",
        "Phorus",
        "pinnacle",
        "PIONEER",
        "Planet Waves",
        "PLX",
        "Polaroid",
        "POLK",
        "Poly-Planar",
        "Powerbass",
        "PowerVision",
        "PreSonus",
        "PRESTIGE",
        "Pro Armor",
        "Pro-Fit",
        "Pro-Ject",
        "PROFILE",
        "Pryma",
        "PSB",
        "PYLE",
        "Python",
        "Q-Form",
        "q-logic",
        "QSC",
        "Questyle Audio",
        "RapcoHorizon",
        "RCA",
        "ReplayTV",
        "Retrosound",
        "Revel",
        "Rhapsody",
        "Ring",
        "Riva",
        "rockfordfosgate",
        "Roku",
        "Rosen",
        "Roth Audio",
        "Russound",
        "Samson",
        "samsung",
        "Sangean",
        "SANUS",
        "SANYO",
        "SAS",
        "SAT RADIO",
        "SCOSCHE",
        "Screen Innovations",
        "Select Intruments",
        "SEMA",
        "Sennheiser",
        "Sensaphone",
        "Sevic",
        "sharp",
        "SHERWOOD",
        "Shure",
        "Simple Control",
        "Singsation",
        "SiriusXM",
        "Slingbox",
        "SlingTV",
        "SOL Republic",
        "Solidsteel",
        "Sonance",
        "Sonos",
        "SONY",
        "Sound Freq",
        "Sound Ordnance",
        "SoundMatters",
        "SoundTube",
        "Spec Tools",
        "SPRINT",
        "Squaretrade",
        "SSV Works",
        "Steinberg",
        "StreetGlow",
        "StreetWires",
        "Sumiko",
        "SunBriteTV",
        "SunFire",
        "SVS",
        "T-Spec",
        "Tamrac",
        "Tamron",
        "TASCAM",
        "TEAC",
        "TechCraft",
        "Technics",
        "Tepui",
        "terk",
        "THIEL",
        "Thinkware",
        "Thule",
        "THX",
        "Tivo",
        "Tivoli",
        "TomTom",
        "Toshiba",
        "Toucan",
        "Trackimo",
        "TrackR",
        "Transcend",
        "TV Networks",
        "ULTIMATE",
        "Ultimate Ears",
        "UpLift",
        "Urbanears",
        "USA Spec",
        "USAcoustics",
        "Usage Guidelines",
        "Utilimedic",
        "V-MODA",
        "Valor",
        "Vanguard Dynamics",
        "varad",
        "Velbon",
        "Velodyne",
        "Ventev",
        "VIP Golf Cars",
        "Viper",
        "VISIONIK",
        "Visualogic",
        "Voxx",
        "W?üvtech",
        "Wadia",
        "Wagen",
        "Waylens",
        "WeatherTech",
        "Weego",
        "WeMo",
        "Wet Sounds",
        "Wharefedale",
        "Whirlwind",
        "WHISTLER",
        "Wi-Ex",
        "Windows Media",
        "Wiremold",
        "Wisenet",
        "Withings",
        "Wonderwoof",
        "Wren",
        "WyreStorm",
        "XM Radio",
        "XS Power",
        "Yak Power",
        "Yale",
        "YAMAHA",
        "Yuneec",
        "ZON",
        "Zoom",
        "ZVOX",
    ],
    Volkswagen: ["Any*", "Beetle", "Golf Alltrack"],
    "Volume Control": ["Any*", "In-wall"],
    Watch: ["Any*", "GPS", "Sports"],
    Wireless: ["Any*", "Range Extender"],
    Wiring: ["Any*", "Harness", "Kit"],
    Years: [
        "Any*",
        "2007",
        "2010",
        "2011",
        "2016",
        "2017",
        "2018",
        "2019",
        "2020",
        "2022",
    ],
};

export const ZoneData = [
    {
        "name": "Z096",
        "isDivider": true,
        "value": ""
    },
    {
        "name": "R101",
        "isDivider": false,
        "value": "F001Z096R101"
    },
    {
        "name": "R102",
        "isDivider": false,
        "value": "F001Z096R102"
    },
    {
        "name": "R103",
        "isDivider": false,
        "value": "F001Z096R103"
    },
    {
        "name": "Z097",
        "isDivider": true,
        "value": ""
    },
    {
        "name": "R101",
        "isDivider": false,
        "value": "F001Z097R101"
    },
    {
        "name": "R102",
        "isDivider": false,
        "value": "F001Z097R102"
    },
    {
        "name": "Z098",
        "isDivider": true,
        "value": ""
    },
    {
        "name": "R101",
        "isDivider": false,
        "value": "F001Z098R101"
    },
    {
        "name": "R102",
        "isDivider": false,
        "value": "F001Z098R102"
    }
]

export const Authors = [
    "Zak Billmeier",
    "Dominic DeVito",
    "Robert Ferency-Viars",
    "Matt Freeman",
    "Julie Govan",
    "Ralph Graves",
    "Steve Kindig",
    "Ken Nail",
    "Charlie Pastorfield",
    "Buck Pomerantz",
    "James Ralston",
    "Tara W.",
    "Kramer Crane",
    "Michael Sokolowski",
    "Jim Richardson",
    "Mike Colley",
    "Jon Paul",
    "David Brown",
    "Dave Delamere",
    "Eric A.",
    "Alexander H.",
    "Bob Bihlmayer",
    "Jeff Miller",
    "Deia Z.",
    "Alexis G",
    "Emily S.",
    "Benn Grant",
    "Colin M.",
    "Ned Oldham",
    "Archer A."
]

export const Facilities = [
    {
        "FacilityIdString": "001",
        "FacilityId": 100,
        "Address": {
            "Name": "Distribution Center",
            "Company": null,
            "Street": "1540 Quiet Acres Lane",
            "Street2": "",
            "City": "Charlottesville",
            "State": "VA",
            "Zip": "229119073",
            "Phone1": "4348171019",
            "Country": "USA"
        },
        "FacilityCode": "CDC",
        "FacilityType": "S",
        "ADSIShipper": "CRUTCH",
        "DefaultShipFacility": true,
        "DefaultPrintFacility": true,
        "TimeZone": null,
        "Properties": [
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 3,
                "PropertyDescription": "First Receipt Capable",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 5,
                "PropertyDescription": "Update WhseFacilityMaster",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 8,
                "PropertyDescription": "Printinv - Printing Enabled",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 10,
                "PropertyDescription": "Package Cubing Enabled",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 11,
                "PropertyDescription": "Printinv - Include Item/Invoice Summaries in Package Group Packet",
                "PropertyType": 3,
                "PropertyValue": false
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 14,
                "PropertyDescription": "Exception Lane",
                "PropertyType": 11,
                "PropertyValue": 99
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 1,
                "PropertyDescription": "LocalTimeZone",
                "PropertyType": 16,
                "PropertyValue": "Eastern Standard Time"
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 4,
                "PropertyDescription": "Printinv - Input Pathname",
                "PropertyType": 16,
                "PropertyValue": "\\\\devdcprintinv\\process$\\Printinv\\DC\\PrintInvInput\\"
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 6,
                "PropertyDescription": "Local Time Zone Abbreviation",
                "PropertyType": 16,
                "PropertyValue": "EST"
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 7,
                "PropertyDescription": "Icon Pathname",
                "PropertyType": 16,
                "PropertyValue": "https://devpix.crutchfield.ad.crutchfield.com/devonly/warehouse-apps/facility-icons/dc.png"
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 9,
                "PropertyDescription": "Printinv - Printed Invoice Types (Describes invoices which generate a postscript file. The Printer Queue Service later sends it to a facility printer.)",
                "PropertyType": 16,
                "PropertyValue": "JiffyBag|SpecialHandling"
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 12,
                "PropertyDescription": "Printinv - Printed Package Group Types (A single postscript file is created for the entire package group instead of many individual invoice postscript files. All invoices in the PG must be a printable invoice type - see Printed Invoice Types.)",
                "PropertyType": 16,
                "PropertyValue": "JiffyBag|SpecialHandling"
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 13,
                "PropertyDescription": "Numina WebService URI",
                "PropertyType": 16,
                "PropertyValue": "https://172.29.3.10:8443/Crutchfield/LabelDownload"
            },
            {
                "FacilityId": 100,
                "PropertyDescriptionId": 15,
                "PropertyDescription": "Printinv - Root Path (subfolder names found in Printinv app.config file)",
                "PropertyType": 16,
                "PropertyValue": "\\\\devdcprintinv\\process$\\PrintInv\\DC\\"
            }
        ],
        "Conditions": [
            {
                "FacilityIdString": "001",
                "ConditionCode": "01",
                "Description": "Ideal sale condition.  No damage.  Not opened",
                "PairQty": 1,
                "SalesBit": 7,
                "Visibility": true,
                "PhysicalInventory": true,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "001",
                "ConditionCode": "02",
                "Description": "Saleable by mail.  No damage.  Opened, but has all contents",
                "PairQty": 1,
                "SalesBit": 7,
                "Visibility": true,
                "PhysicalInventory": true,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "001",
                "ConditionCode": "03",
                "Description": "Unknown condition.  For evaluation by Testing",
                "PairQty": 2,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": false,
                "AdjustStockForReturns": true,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "001",
                "ConditionCode": "04",
                "Description": "Return to Vendor.  Defects that warrant return to vendor",
                "PairQty": 2,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "001",
                "ConditionCode": "05",
                "Description": "Saleable by retail stores only.  Some defects.  Opened.  Cus",
                "PairQty": 1,
                "SalesBit": 1,
                "Visibility": true,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "001",
                "ConditionCode": "06",
                "Description": "Incomplete.",
                "PairQty": 2,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "001",
                "ConditionCode": "07",
                "Description": "Scratch and Dent.  Small defects.  Opened.  Saleable",
                "PairQty": 1,
                "SalesBit": 7,
                "Visibility": true,
                "PhysicalInventory": true,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "001",
                "ConditionCode": "08",
                "Description": "Display item.",
                "PairQty": 2,
                "SalesBit": 0,
                "Visibility": false,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "001",
                "ConditionCode": "09",
                "Description": "Drop Ship. Same as 01",
                "PairQty": 1,
                "SalesBit": 7,
                "Visibility": false,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": true
            }
        ]
    },
    {
        "FacilityIdString": "002",
        "FacilityId": 101,
        "Address": {
            "Name": "Reas Ford",
            "Company": null,
            "Street": "395 Reas Ford Rd STE 170",
            "Street2": "",
            "City": "Earlysville",
            "State": "VA",
            "Zip": "229362462",
            "Phone1": "4348171019",
            "Country": "USA"
        },
        "FacilityCode": "RFC",
        "FacilityType": "S",
        "ADSIShipper": "REASFORD",
        "DefaultShipFacility": false,
        "DefaultPrintFacility": false,
        "TimeZone": null,
        "Properties": [
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 2,
                "PropertyDescription": "CA Proposition 65 Capable",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 3,
                "PropertyDescription": "First Receipt Capable",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 5,
                "PropertyDescription": "Update WhseFacilityMaster",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 8,
                "PropertyDescription": "Printinv - Printing Enabled",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 10,
                "PropertyDescription": "Package Cubing Enabled",
                "PropertyType": 3,
                "PropertyValue": false
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 11,
                "PropertyDescription": "Printinv - Include Item/Invoice Summaries in Package Group Packet",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 14,
                "PropertyDescription": "Exception Lane",
                "PropertyType": 11,
                "PropertyValue": 99
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 1,
                "PropertyDescription": "LocalTimeZone",
                "PropertyType": 16,
                "PropertyValue": "Eastern Standard Time"
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 4,
                "PropertyDescription": "Printinv - Input Pathname",
                "PropertyType": 16,
                "PropertyValue": "\\\\devdcprintinv\\process$\\Printinv\\DC\\PrintInvInput\\"
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 6,
                "PropertyDescription": "Local Time Zone Abbreviation",
                "PropertyType": 16,
                "PropertyValue": "EST"
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 7,
                "PropertyDescription": "Icon Pathname",
                "PropertyType": 16,
                "PropertyValue": "https://devpix.crutchfield.ad.crutchfield.com/devonly/warehouse-apps/facility-icons/reas-ford.png"
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 9,
                "PropertyDescription": "Printinv - Printed Invoice Types (Describes invoices which generate a postscript file. The Printer Queue Service later sends it to a facility printer.)",
                "PropertyType": 16,
                "PropertyValue": "JiffyBag|SpecialHandling"
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 12,
                "PropertyDescription": "Printinv - Printed Package Group Types (A single postscript file is created for the entire package group instead of many individual invoice postscript files. All invoices in the PG must be a printable invoice type - see Printed Invoice Types.)",
                "PropertyType": 16,
                "PropertyValue": "JiffyBag|SpecialHandling"
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 13,
                "PropertyDescription": "Numina WebService URI",
                "PropertyType": 16,
                "PropertyValue": "https://172.29.3.10:8443/Crutchfield/LabelDownload"
            },
            {
                "FacilityId": 101,
                "PropertyDescriptionId": 15,
                "PropertyDescription": "Printinv - Root Path (subfolder names found in Printinv app.config file)",
                "PropertyType": 16,
                "PropertyValue": "\\\\devdcprintinv\\process$\\Printinv\\DC\\"
            }
        ],
        "Conditions": [
            {
                "FacilityIdString": "002",
                "ConditionCode": "01",
                "Description": "Ideal sale condition.  No damage.  Not opened",
                "PairQty": 1,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "002",
                "ConditionCode": "02",
                "Description": "Saleable by mail.  No damage.  Opened, but has all contents",
                "PairQty": 1,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "002",
                "ConditionCode": "03",
                "Description": "Unknown condition.  For evaluation by Testing",
                "PairQty": 2,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": true,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "002",
                "ConditionCode": "04",
                "Description": "Return to Vendor.  Defects that warrant return to vendor",
                "PairQty": 2,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": true,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "002",
                "ConditionCode": "05",
                "Description": "Saleable by retail stores only.  Some defects.  Opened.  Cus",
                "PairQty": 1,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": true,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "002",
                "ConditionCode": "06",
                "Description": "Incomplete.",
                "PairQty": 2,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": true,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "002",
                "ConditionCode": "07",
                "Description": "Scratch and Dent.  Small defects.  Opened.  Saleable",
                "PairQty": 1,
                "SalesBit": 0,
                "Visibility": true,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "002",
                "ConditionCode": "08",
                "Description": "Display item.",
                "PairQty": 2,
                "SalesBit": 0,
                "Visibility": false,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": false
            },
            {
                "FacilityIdString": "002",
                "ConditionCode": "09",
                "Description": "Drop Ship. Same as 01",
                "PairQty": 1,
                "SalesBit": 0,
                "Visibility": false,
                "PhysicalInventory": false,
                "AdjustStockForReturns": false,
                "ShipsFromVendor": true
            }
        ]
    },
    {
        "FacilityIdString": "003",
        "FacilityId": 109,
        "Address": {
            "Name": "Phoenix DC",
            "Company": null,
            "Street": "1234 Unknown St",
            "Street2": "",
            "City": "Phoenix",
            "State": "NV",
            "Zip": "850331717",
            "Phone1": "7029041000",
            "Country": "USA"
        },
        "FacilityCode": "PDC",
        "FacilityType": "S",
        "ADSIShipper": "PHOENIX",
        "DefaultShipFacility": false,
        "DefaultPrintFacility": false,
        "TimeZone": null,
        "Properties": [
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 2,
                "PropertyDescription": "CA Proposition 65 Capable",
                "PropertyType": 3,
                "PropertyValue": false
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 3,
                "PropertyDescription": "First Receipt Capable",
                "PropertyType": 3,
                "PropertyValue": false
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 5,
                "PropertyDescription": "Update WhseFacilityMaster",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 8,
                "PropertyDescription": "Printinv - Printing Enabled",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 10,
                "PropertyDescription": "Package Cubing Enabled",
                "PropertyType": 3,
                "PropertyValue": false
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 11,
                "PropertyDescription": "Printinv - Include Item/Invoice Summaries in Package Group Packet",
                "PropertyType": 3,
                "PropertyValue": true
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 1,
                "PropertyDescription": "LocalTimeZone",
                "PropertyType": 16,
                "PropertyValue": "US Mountain Standard Time"
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 4,
                "PropertyDescription": "Printinv - Input Pathname",
                "PropertyType": 16,
                "PropertyValue": "\\\\devdcprintinv\\process$\\PrintInv\\DC\\PrintInvInput\\"
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 6,
                "PropertyDescription": "Local Time Zone Abbreviation",
                "PropertyType": 16,
                "PropertyValue": "MST"
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 7,
                "PropertyDescription": "Icon Pathname",
                "PropertyType": 16,
                "PropertyValue": "https://devpix.crutchfield.ad.crutchfield.com/devonly/warehouse-apps/facility-icons/phoenix.png"
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 9,
                "PropertyDescription": "Printinv - Printed Invoice Types (Describes invoices which generate a postscript file. The Printer Queue Service later sends it to a facility printer.)",
                "PropertyType": 16,
                "PropertyValue": "JiffyBag|SpecialHandling"
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 12,
                "PropertyDescription": "Printinv - Printed Package Group Types (A single postscript file is created for the entire package group instead of many individual invoice postscript files. All invoices in the PG must be a printable invoice type - see Printed Invoice Types.)",
                "PropertyType": 16,
                "PropertyValue": "JiffyBag|SpecialHandling"
            },
            {
                "FacilityId": 109,
                "PropertyDescriptionId": 15,
                "PropertyDescription": "Printinv - Root Path (subfolder names found in Printinv app.config file)",
                "PropertyType": 16,
                "PropertyValue": "\\\\devdcprintinv\\process$\\Printinv\\DC\\"
            }
        ],
        "Conditions": []
    }
]

export const ClientFacility = {
    "FacilityIdString": "003",
    "FacilityId": 109,
    "Address": {
        "Name": "Phoenix DC",
        "Company": null,
        "Street": "1234 Unknown St",
        "Street2": "",
        "City": "Phoenix",
        "State": "NV",
        "Zip": "850331717",
        "Phone1": "7029041000",
        "Country": "USA"
    },
    "FacilityCode": "PDC",
    "FacilityType": "S",
    "ADSIShipper": "PHOENIX",
    "DefaultShipFacility": false,
    "DefaultPrintFacility": false,
    "TimeZone": {
        "Id": "US Mountain Standard Time",
        "DisplayName": "(UTC-07:00) Arizona",
        "StandardName": "US Mountain Standard Time",
        "DaylightName": "US Mountain Daylight Time",
        "BaseUtcOffset": "-07:00:00",
        "AdjustmentRules": null,
        "SupportsDaylightSavingTime": false
    },
    "Properties": [
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 2,
            "PropertyDescription": null,
            "PropertyType": 3,
            "PropertyValue": false
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 3,
            "PropertyDescription": null,
            "PropertyType": 3,
            "PropertyValue": false
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 5,
            "PropertyDescription": null,
            "PropertyType": 3,
            "PropertyValue": true
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 8,
            "PropertyDescription": null,
            "PropertyType": 3,
            "PropertyValue": true
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 10,
            "PropertyDescription": null,
            "PropertyType": 3,
            "PropertyValue": false
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 11,
            "PropertyDescription": null,
            "PropertyType": 3,
            "PropertyValue": true
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 1,
            "PropertyDescription": null,
            "PropertyType": 16,
            "PropertyValue": "US Mountain Standard Time"
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 4,
            "PropertyDescription": null,
            "PropertyType": 16,
            "PropertyValue": "\\\\devdcprintinv\\process$\\PrintInv\\DC\\PrintInvInput\\"
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 6,
            "PropertyDescription": null,
            "PropertyType": 16,
            "PropertyValue": "MST"
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 7,
            "PropertyDescription": null,
            "PropertyType": 16,
            "PropertyValue": "https://devpix.crutchfield.ad.crutchfield.com/devonly/warehouse-apps/facility-icons/phoenix.png"
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 9,
            "PropertyDescription": null,
            "PropertyType": 16,
            "PropertyValue": "JiffyBag|SpecialHandling"
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 12,
            "PropertyDescription": null,
            "PropertyType": 16,
            "PropertyValue": "JiffyBag|SpecialHandling"
        },
        {
            "FacilityId": 109,
            "PropertyDescriptionId": 15,
            "PropertyDescription": null,
            "PropertyType": 16,
            "PropertyValue": "\\\\devdcprintinv\\process$\\Printinv\\DC\\"
        }
    ],
    "Conditions": []
}

/**
 * Data grid
 * */
const DiscrepanciesData = {
    "transferContainerId": 0,
    "containerId": 0,
    "sourceFacilityId": 0,
    "destinationFacilityId": 0,
    "transferContainerConditions": [],
    "transferContainerZones": [],
    "transferItems": [
        {
            "transferContainerItemId": 10,
            "tranCode": 0,
            "itemNo": "305LSP7T",
            "containerId": 0,
            "sourceQty": 1,
            "destinationQty": 6,
            "protectedSourceQty": 0,
            "sourceCondition": {
                "conditionCode": "03",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "destinationCondition": {
                "conditionCode": "02",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "sourceFacilityId": 101,
            "destinationFacilityId": 101,
            "itemDescription": "Samsung LSP7T The Premiere Projector",
            "itemCost": 2368.08,
            "processedByEmpId": 3842,
            "dateProcessed": "2021-07-23T05:28:00-04:00",
            "assignedByEmpId": 0,
            "dateAssigned": "0001-01-01T00:00:00+00:00",
            "transferContainerItemStatusDescId": 0,
            "transferContainerItemStatus": "",
            "transferContainerItemStatusDesc": ""
        },
        {
            "transferContainerItemId": 11,
            "tranCode": 0,
            "itemNo": "111111X",
            "containerId": 0,
            "sourceQty": 1,
            "destinationQty": 5,
            "protectedSourceQty": 0,
            "sourceCondition": {
                "conditionCode": "07",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "destinationCondition": {
                "conditionCode": "07",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "sourceFacilityId": 101,
            "destinationFacilityId": 100,
            "itemDescription": "LG TV",
            "itemCost": 2368.08,
            "processedByEmpId": 3842,
            "dateProcessed": "2021-07-23T05:28:00-04:00",
            "assignedByEmpId": 0,
            "dateAssigned": "0001-01-01T00:00:00+00:00",
            "transferContainerItemStatusDescId": 0,
            "transferContainerItemStatus": "",
            "transferContainerItemStatusDesc": ""
        },
        {
            "transferContainerItemId": 12,
            "tranCode": 0,
            "itemNo": "XT323234",
            "containerId": 0,
            "sourceQty": 1,
            "destinationQty": 6,
            "protectedSourceQty": 0,
            "sourceCondition": {
                "conditionCode": "03",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "destinationCondition": {
                "conditionCode": "02",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "sourceFacilityId": 101,
            "destinationFacilityId": 101,
            "itemDescription": "Apple TV",
            "itemCost": 2368.08,
            "processedByEmpId": 3830,
            "dateProcessed": "2021-07-23T05:28:00-04:00",
            "assignedByEmpId": 0,
            "dateAssigned": "0001-01-01T00:00:00+00:00",
            "transferContainerItemStatusDescId": 0,
            "transferContainerItemStatus": "",
            "transferContainerItemStatusDesc": ""
        },
        {
            "transferContainerItemId": 13,
            "tranCode": 0,
            "itemNo": "2900XRD",
            "containerId": 0,
            "sourceQty": 1,
            "destinationQty": 5,
            "protectedSourceQty": 0,
            "sourceCondition": {
                "conditionCode": "03",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "destinationCondition": {
                "conditionCode": "02",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "sourceFacilityId": 101,
            "destinationFacilityId": 100,
            "itemDescription": "Speaker product",
            "itemCost": 2368.08,
            "processedByEmpId": 3842,
            "dateProcessed": "2021-07-23T05:28:00-04:00",
            "assignedByEmpId": 0,
            "dateAssigned": "0001-01-01T00:00:00+00:00",
            "transferContainerItemStatusDescId": 0,
            "transferContainerItemStatus": "",
            "transferContainerItemStatusDesc": ""
        },
        {
            "transferContainerItemId": 5828,
            "tranCode": 0,
            "itemNo": "280REBELT3",
            "containerId": 0,
            "sourceQty": 1,
            "destinationQty": 2,
            "protectedSourceQty": 0,
            "sourceCondition": {
                "conditionCode": "03",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "destinationCondition": {
                "conditionCode": "02",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "sourceFacilityId": 101,
            "destinationFacilityId": 100,
            "itemDescription": "Canon EOS Rebel T3 Kit w/ 18-55mm Lens",
            "itemCost": 372.28,
            "processedByEmpId": 2112,
            "dateProcessed": "2022-10-17T11:36:55.58-04:00",
            "assignedByEmpId": 0,
            "dateAssigned": "0001-01-01T00:00:00+00:00",
            "transferContainerItemStatusDescId": 0,
            "transferContainerItemStatus": "",
            "transferContainerItemStatusDesc": ""
        },
        {
            "transferContainerItemId": 5829,
            "tranCode": 0,
            "itemNo": "280REBELT3",
            "containerId": 0,
            "sourceQty": 1,
            "destinationQty": 1,
            "protectedSourceQty": 0,
            "sourceCondition": {
                "conditionCode": "03",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "destinationCondition": {
                "conditionCode": "02",
                "pairQty": 0,
                "salesBit": 0,
                "visibility": false,
                "physicalInventory": false,
                "adjustStockForReturns": false,
                "shipsFromVendor": false
            },
            "sourceFacilityId": 101,
            "destinationFacilityId": 101,
            "itemDescription": "Canon EOS Rebel T3 Kit w/ 18-55mm Lens",
            "itemCost": 372.28,
            "processedByEmpId": 2112,
            "dateProcessed": "2022-10-17T11:37:58.473-04:00",
            "assignedByEmpId": 0,
            "dateAssigned": "0001-01-01T00:00:00+00:00",
            "transferContainerItemStatusDescId": 0,
            "transferContainerItemStatus": "",
            "transferContainerItemStatusDesc": ""
        }
    ],
    "dateCreated": "0001-01-01T00:00:00+00:00",
    "lastActionDateTime": "0001-01-01T00:00:00+00:00",
    "lastActionByEmpId": 0,
    "transferContainerStatusId": 0,
    "transferContainerStatus": "",
    "transferContainerStatusDesc": "",
    "transferContainerTypeId": 0,
    "transferContainerType": "",
    "transferContainerTypeDesc": "",
    "totalItemCost": 3112,
    "totalItemCount": 3,
    "allowContainerCloseAction": false,
    "allowContainerDeleteAction": false
}

// create a header object that will display your headers in the data grid
export const dataGridHeaders = {
    'ItemNo': {
        'headerName': 'Item No',
        'propertyName': 'itemNo'
    },
    'Description': {
        'headerName': 'Description',
        'propertyName': 'description'
    },
    'DestinationQty': {
        'headerName': 'Source Qty',
        'propertyName': 'destinationQty'
    },
    'sourceCondition': {
        'headerName': 'From Container',
        'propertyName': 'sourceCondition'
    },
    'destinationCondition': {
        'headerName': 'To Container',
        'propertyName': 'destinationCondition'
    },
    'sourceFacilityId': {
        'headerName': 'From Facility',
        'propertyName': 'sourceFacilityId'
    },
    'destinationFacilityId': {
        'headerName': 'To Facility',
        'propertyName': 'destinationFacilityId'
    },
    'EmployeeId': {
        'headerName': 'EmployeeId',
        'propertyName': 'processedByEmpId'
    },
    'DateProcessed': {
        'headerName': 'Date Processed',
        'propertyName': 'dateProcessed'
    }
}

const facilityAbbreviation = (facilityId: number) => {
    switch (facilityId) {
        case 100:
            return 'DC';
        case 101:
            return 'RPC';
    }
    return null;
};
// create a new object of all of the properties you want to have displayed on the data grid
export const dataGridRows = ref(DiscrepanciesData.transferItems.map((item) => {
    return {
        'transferContainerItemId': {
            name: item.transferContainerItemId,
            display: false,
            link: ''
        },
        'itemNo': {
            name: item.itemNo,
            display: true,
            link: 'http://sqlreportserv/Reports/report/Open/Warehouse/WarehouseTransferOutofTesting'
        },
        'description': {
            name: item.itemDescription,
            display: true,
            link: ''
        },
        'destinationQty': {
            name: item.destinationQty,
            display: true
        },
        'sourceCondition': {
            name: item.sourceCondition.conditionCode,
            display: true
        },
        'destinationCondition': {
            name: item.destinationCondition.conditionCode,
            display: true
        },
        'sourceFacilityId': {
            name: item.sourceFacilityId,
            display: true,
            link: '',
            hasToolTip: true,
            toolTipText: facilityAbbreviation(item.sourceFacilityId)
        },
        'destinationFacilityId': {
            name: item.destinationFacilityId,
            display: true
        },
        'processedByEmpId': {
            name: item.processedByEmpId,
            display: true
        },
        'dateProcessed': {
            name: formatDateMMDDYYYY(item.dateProcessed),
            display: true
        },
        'currentSort': '',
        'currentSortDir': 'asc'
    }
}));

const locationHistoryData = ref([
    {
        "activityDate": "2022-10-31T13:26:06.13",
        "user": "KBARRETT",
        "program": "RTV32",
        "condition": "04",
        "location": {
            "locId": 14397,
            "facility": "002",
            "zone": "TRN",
            "row": "000",
            "stack": "000",
            "level": "000",
            "bin": "000",
            "isPickable": false,
            "isSuggestion": false,
            "stackTypeId": 0,
            "pickTypeId": 0,
            "facilityId": 0
        },
        "qtyOld": 0,
        "qtyNew": 0,
        "qtyChanged": 0,
        "containerId": "None",
        "itemNo": "714FIVESWN"
    },
    {
        "activityDate": "2022-10-31T13:26:06.123",
        "user": "KBARRETT",
        "program": "RTV32",
        "condition": "04",
        "location": {
            "locId": 14397,
            "facility": "002",
            "zone": "TRN",
            "row": "000",
            "stack": "000",
            "level": "000",
            "bin": "000",
            "isPickable": false,
            "isSuggestion": false,
            "stackTypeId": 0,
            "pickTypeId": 0,
            "facilityId": 0
        },
        "qtyOld": 1,
        "qtyNew": 0,
        "qtyChanged": -1,
        "containerId": "None",
        "itemNo": "714FIVESWN"
    },
    {
        "activityDate": "2022-10-28T11:02:52.397",
        "user": "WDOWDELL",
        "program": "Warehouse Transfer",
        "condition": "04",
        "location": {
            "locId": 14397,
            "facility": "DCR",
            "zone": "TRN",
            "row": "000",
            "stack": "000",
            "level": "000",
            "bin": "000",
            "isPickable": false,
            "isSuggestion": false,
            "stackTypeId": 0,
            "pickTypeId": 0,
            "facilityId": 0
        },
        "qtyOld": 0,
        "qtyNew": 1,
        "qtyChanged": 1,
        "containerId": "None",
        "itemNo": "714FIVESWN"
    },
    {
        "activityDate": "2022-09-22T00:12:09.943",
        "user": "JBOYD",
        "program": "OffsiteScan",
        "condition": "03",
        "location": {
            "locId": 109982,
            "facility": "002",
            "zone": "070",
            "row": "047",
            "stack": "010",
            "level": "002",
            "bin": "001",
            "isPickable": false,
            "isSuggestion": false,
            "stackTypeId": 0,
            "pickTypeId": 0,
            "facilityId": 0
        },
        "qtyOld": 1,
        "qtyNew": 0,
        "qtyChanged": -1,
        "containerId": "C7907739",
        "itemNo": "714FIVESWN"
    },
    {
        "activityDate": "2022-09-22T00:05:43.127",
        "user": "JBOYD",
        "program": "OffsiteScan",
        "condition": "03",
        "location": {
            "locId": 109982,
            "facility": "002",
            "zone": "070",
            "row": "047",
            "stack": "010",
            "level": "002",
            "bin": "001",
            "isPickable": false,
            "isSuggestion": false,
            "stackTypeId": 0,
            "pickTypeId": 0,
            "facilityId": 0
        },
        "qtyOld": 0,
        "qtyNew": 1,
        "qtyChanged": 1,
        "containerId": "C7907739",
        "itemNo": "714FIVESWN"
    },
    {
        "activityDate": "2022-09-08T19:39:58.513",
        "user": "TBRIZZOLARA",
        "program": "OffsiteScan",
        "condition": "07",
        "location": {
            "locId": 114502,
            "facility": "002",
            "zone": "070",
            "row": "046",
            "stack": "007",
            "level": "003",
            "bin": "000",
            "isPickable": false,
            "isSuggestion": false,
            "stackTypeId": 0,
            "pickTypeId": 0,
            "facilityId": 0
        },
        "qtyOld": 0,
        "qtyNew": 1,
        "qtyChanged": 1,
        "containerId": "None",
        "itemNo": "714FIVESWN"
    },
    {
        "activityDate": "2022-09-08T19:39:58.433",
        "user": "TBRIZZOLARA",
        "program": "OffsiteScan",
        "condition": "03",
        "location": {
            "locId": 40039,
            "facility": "DCR",
            "zone": "070",
            "row": "044",
            "stack": "017",
            "level": "001",
            "bin": "001",
            "isPickable": false,
            "isSuggestion": false,
            "stackTypeId": 0,
            "pickTypeId": 0,
            "facilityId": 0
        },
        "qtyOld": 0,
        "qtyNew": 0,
        "qtyChanged": 0,
        "containerId": "C7878730",
        "itemNo": "714FIVESWN"
    },
    {
        "activityDate": "2022-09-08T18:59:52.963",
        "user": "TBRIZZOLARA",
        "program": "OffsiteScan",
        "condition": "03",
        "location": {
            "locId": 40039,
            "facility": "002",
            "zone": "070",
            "row": "044",
            "stack": "017",
            "level": "001",
            "bin": "001",
            "isPickable": false,
            "isSuggestion": false,
            "stackTypeId": 0,
            "pickTypeId": 0,
            "facilityId": 0
        },
        "qtyOld": 0,
        "qtyNew": 1,
        "qtyChanged": 1,
        "containerId": "C7878730",
        "itemNo": "714FIVESWN"
    },

])
// create a header object that will display your headers in the data grid
export const dataGridHeadersCustomSort = {
    'activityDate:': {
        'propertyName': 'activityDate',
        'headerName': 'Date & Time'
    },
    'user': {
        'propertyName': 'user',
        'headerName': 'User'
    },
    'program': {
        'propertyName': 'program',
        'headerName': 'Program'
    },
    'condition': {
        'propertyName': 'condition',
        'headerName': 'Condition'
    },
    'fac': {
        'propertyName': 'fac',
        'headerName': 'Fac'
    },
    'zone': {
        'propertyName': 'zone',
        'headerName': 'Zone'
    },
    'row': {
        'propertyName': 'row',
        'headerName': 'Row'
    },
    'stack': {
        'propertyName': 'stack',
        'headerName': 'Stack'
    },
    'level': {
        'propertyName': 'level',
        'headerName': 'Level'
    },
    'bin': {
        'propertyName': 'bin',
        'headerName': 'Bin'
    },
    'qtyOld': {
        'propertyName': 'qtyOld',
        'headerName': 'Qty'
    },
    'qtyChanged': {
        'propertyName': 'qtyChanged',
        'headerName': 'Qty Change'
    },
    'qtyNew': {
        'propertyName': 'qtyNew',
        'headerName': 'Qty New'
    },
    'containerId': {
        'propertyName': 'containerId',
        'headerName': 'Container'
    },
    'itemNo': {
        'propertyName': 'itemNo',
        'headerName': 'Item No'
    }
}

// create a new object of all of the properties you want to have displayed on the data grid
export const dataGridRowsCustomSort = ref(locationHistoryData.value.map((tableEntry) => {
        return {
            'activityDate': {
                name: formatDateMMDDYYYY(tableEntry.activityDate),
                display: true,
                link: '',
            },
            'user': {
                name: tableEntry.user,
                display: true,
                link: 'http://intranet.crutchfield.ad.crutchfield.com/cgi-bin/warehouse/ifinder/ifinder.aspx?Item=714FIVESWN&FacilityId=002&Condition=All',
            },
            'program': {
                name: tableEntry.program,
                display: true,
                link: ''
            },
            'condition': {
                name: tableEntry.condition,
                display: true,
                link: ''
            },
            'fac': {
                name: tableEntry.location.facility,
                display: true,
                link: ''
            },
            'zone': {
                name: tableEntry.location.zone,
                display: true,
                link: ''
            },
            'row': {
                name: tableEntry.location.row,
                display: true,
                link: ''
            },
            'stack': {
                name: tableEntry.location.stack,
                display: true,
                link: ''
            },
            'level': {
                name: tableEntry.location.level,
                display: true,
                link: ''
            },
            'bin': {
                name: tableEntry.location.bin,
                display: true,
                link: ''
            },
            'qtyOld': {
                name: tableEntry.qtyOld,
                display: true,
                link: ''
            },
            'qtyChanged': {
                name: tableEntry.qtyChanged,
                display: true,
                link: '',
                cellStyling: qtyChange(tableEntry.qtyChanged)
            },
            'qtyNew': {
                name: tableEntry.qtyNew,
                display: true,
                link: ''
            },
            'containerId': {
                name: tableEntry.containerId,
                display: true,
                link: tableEntry.containerId !== 'None' ? 'http://intranet.crutchfield.ad.crutchfield.com/cgi-bin/warehouse/ifinder/ifinder.aspx?Item=714FIVESWN&FacilityId=002&Condition=All' : '',
            },
            'itemNo': {
                name: tableEntry.itemNo,
                display: true,
                link: 'http://intranet.crutchfield.ad.crutchfield.com/cgi-bin/warehouse/ifinder/ifinder.aspx?Item=714FIVESWN&FacilityId=002&Condition=All',
            },
            'currentSort': '',
            'currentSortDir': 'asc',
            'customRowStyling': tableEntry.location.facility === 'DCR' ? 'discrepancy' : '',
        }
})
    )

function qtyChange(qty: number) {
    let qtyChange;
    if (qty === 0) {
        return qtyChange = '';
    } else {
        qtyChange = Math.sign(qty) === -1 ? 'negative-qty' : 'positive-qty';
        return qtyChange;
    }
}
